source/memtrace.c

```diff
@@ -483,35 +498,37 @@ struct aws_allocator *aws_mem_tracer_new(
 
 struct aws_allocator *aws_mem_tracer_destroy(struct aws_allocator *trace_allocator) {
     struct alloc_tracer *tracer = trace_allocator->impl;
-    struct aws_allocator *allocator = tracer->allocator;
-
-    /* This is not necessary, as if you are destroying the allocator, what are your
-     * expectations? Either way, we can, so we might as well...
-     */
-    aws_mutex_lock(&tracer->mutex);
-    aws_hash_table_clean_up(&tracer->allocs);
-    aws_hash_table_clean_up(&tracer->stacks);
-    aws_mutex_unlock(&tracer->mutex);
-    aws_mutex_clean_up(&tracer->mutex);
+    struct aws_allocator *allocator = tracer->traced_allocator;
+
+    if (tracer->level != AWS_MEMTRACE_NONE) {
+        aws_mutex_lock(&tracer->mutex);
+        aws_hash_table_clean_up(&tracer->allocs);
+        aws_hash_table_clean_up(&tracer->stacks);
+        aws_mutex_unlock(&tracer->mutex);
+        aws_mutex_clean_up(&tracer->mutex);
+    }
 
-    struct aws_allocator *system_allocator = tracer->system_allocator;
-    aws_mem_release(system_allocator, tracer);
+    aws_mem_release(tracer->system_allocator, tracer);
     /* trace_allocator is freed as part of the block tracer was allocated in */
-    return allocator;
-}
 
-void aws_mem_tracer_dump(struct aws_allocator *trace_allocator) {
```

Comment:
Was this delete intended?

Reply:
yeah, that was intentional.

`aws_mem_tracer_dump()` was just calling `s_alloc_tracer_dump()`.
I renamed `s_alloc_tracer_dump()` to `aws_mem_tracer_dump()` (higher up in the file) and removed useless indirection down here.